pkg/clustermesh/operator/clustermesh.go

```diff
@@ -79,33 +67,20 @@ type ClusterMesh interface {
 }
 
 func newClusterMesh(lc cell.Lifecycle, params clusterMeshParams) (*clusterMesh, ClusterMesh) {
-	if !params.Clientset.IsEnabled() || params.ClusterMeshConfig == "" || !params.Cfg.ClusterMeshEnableEndpointSync {
+	if params.ClusterMeshConfig == "" || !params.Cfg.ClusterMeshEnableEndpointSync {
```

Comment:
Out of curiosity, what is your plan here with regards to guarding the support for mcsapi?

Reply:
Probably running the `newClusterMesh` if either flag is enabled + passing the boolean to `remote_cluster.go` to register only what's needed

Reply:
sgtm :+1: